guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,53 @@
+# Motivation
+Our changelogs (reviewed by users/customers) are autogenerated from PR titles and git commit text.  The better this information is, the more we help our customers.  This information also helps reviewers understand changes, patch managers understand risk/importance, and when hunting for regressions, helps us better understand what is/isn’t likely to be a cause.
+
+# PR Metadata
+
+__PR title:__ Bug NNNN: A short sentence with capitalization and no period or the Jira tag for the issue “Feature TEAM-XXX:”
+
+* Try to describe the problem the user sees, rather than the code change you are doing
+  * I.e. “Console does not work when user clicks X” instead of “Fix drawer notification integration with underlying controller”
+* Prefix bug fixes with the bug number, “Bug XXXXX:” 
+  * (capitalize bug, space, number, then colon)
+* Do not let your PR title get truncated/wrapped into the description
+* Additional prefixes like what release the PR applies to are allowed, but they will be stripped from the changelog (the changelog for 4.4 doesn’t need to say “release-4.4” 
+  * Changelog strips anything between square brackets that has no spaces - i.e. [release-4.4] is stripped but [release 4.4] would not be
+    
+
+__PR description:__  Additional context reviewers need to understand your PR
+
+* Why was this change needed?
+* What changes are being made?  
+* Reference to any prior work(other PRs) or follow up work that will be needed.
```

Comment:
In general, I try to make my commits as meaningful as possible. Often like an email where the first line is the Subject. The description is more details about what is going on in the PR, motivations links to other related works, etc.

Reply:
then you don't need this doc because you're doing it right already :)